Main.elm

```diff
@@ -14,52 +15,65 @@ import Json.Encode
 type Msg
     = NewEmail String
     | NewPassphrase String
-    | NewData String
+    | NewData (Maybe String)
+    | UpdateContent String
     | NewError String
     | GetData
     | Lock
-    | SetData String
     | DataSaved String
     | DataNotSaved String
-    | UnStored Time.Time
+    | Modifying Time.Time
```

Comment:
A one-line comment saying something like `-- NewData -> UpdateContent -> Modifying -> setData -> DataSaved` would be helpful ;)